studio/components/interfaces/Reports/ReportFilterBar.tsx

```diff
@@ -0,0 +1,286 @@
+        key: nextProductFilter.filterKey,
+        compare: 'matches',
+        value: nextProductFilter.filterValue,
+      })
+    }
+    setCurrentProductFilter(nextProductFilter)
+  }
+
+  return (
+    <div>
+      <div className="flex flex-row justify-start items-center flex-wrap gap-2">
+        <DatePickers
+          onChange={onDatepickerChange}
+          to={datepickerTo}
+          from={datepickerFrom}
+          helpers={datepickerHelpers}
+        />
+        <Dropdown
+          size="small"
+          side="bottom"
+          align="start"
+          overlay={
+            <>
+              <Dropdown.Item onClick={() => handleProductFilterChange(null)}>
+                All Requests
+              </Dropdown.Item>
+              <Dropdown.Separator />
+              {PRODUCT_FILTERS.map((productFilter) => {
+                const Icon = productFilter.icon
+                return (
+                  <Dropdown.Item
+                    key={productFilter.key}
+                    disabled={productFilter.key === currentProductFilter?.key}
+                    onClick={() => handleProductFilterChange(productFilter)}
+                    className="hover:bg-scale-600"
+                    icon={<Icon size={24} />}
+                  >
+                    <span
+                      className={[
+                        productFilter.key === currentProductFilter?.key ? 'font-bold' : '',
+                        'inline-block',
+                      ].join(' ')}
+                    >
+                      {productFilter.label}
+                    </span>
+                    <span className=" text-left text-scale-1000 inline-block">
+                      {productFilter.description}
+                    </span>
+                  </Dropdown.Item>
+                )
+              })}
+            </>
+          }
+        >
+          <Button
+            as="span"
+            type="default"
+            className="inline-flex flex-row gap-2"
+            iconRight={<IconChevronDown size={14} />}
+          >
+            {currentProductFilter === null ? 'All Requests' : currentProductFilter.label}
+          </Button>
+        </Dropdown>
+        {filters
+          .filter(
+            (filter) =>
+              filter.value !== currentProductFilter?.filterValue ||
+              filter.key !== currentProductFilter?.filterKey
+          )
+          .map((filter) => (
+            <div
+              key={`${filter.key}-${filter.compare}-${filter.value}`}
+              className="text-xs rounded border border-scale-1100 bg-scale-500 px-1 h-7 flex flex-row justify-center gap-1 items-center"
+            >
+              {filter.key} {filter.compare} {filter.value}
+              <Button
+                type="text"
+                size="tiny"
+                className="!p-0 h-6 w-6 flex flex-row justify-center items-center"
+                onClick={() => {
+                  onRemoveFilters([filter])
```

Comment:
should users be allowed to add the exact same filter? currently the UI allows it but if i try to remove one of the filter, all of them are removed
![image](https://user-images.githubusercontent.com/19742402/220249665-79d2bae0-5a64-49f8-bb5c-e94a342f6c13.png)

i'd say lets add some validation to prevent adding the same filter

Reply:
👍